### DISCUSSION TITLE
Room Assistant companion constantly running?

### DISCUSSION THREAD

**Discussion by gmalbert** [mKeRix/room-assistant#496] [Q&A] [answered]
Does the room assistant companion app run all the time? From the documentation, it sounds like BLE advertisements are happening constantly without a performance hit (hence the current lack of a BLE switch), but it wasn't clear to me whether the companion app ran in the background or did anything else if I wasn't in range of an RA instance. I obviously don't want it to do any sort of excess battery drain, but I wasn't clear on the design. Any thoughts?

**Comment by mKeRix** [comment, marked as answer]
The app isn't running in the background per se, as it just uses the iOS APIs to tag along with what the iPhone sends regardless. It should cause little to no extra battery drain on the device. My iPhone also doesn't list any battery usage for the room-assistant companion app in its settings.

One additional remark to this question: currently the app is always adding its service to the advertisements - this means that another entity besides you could technically pull your unique "room-assistant ID" from the device to circumvent the privacy feature of changing addresses. I plan to use geofencing or something like that to disable this as soon as you are out of reach of your home in the future. I don't think this is a big concern atm, as the tracking solutions out there likely won't implement this for anything (and there are other methods of fingerprinting), but it's good to be aware of it regardless.

**Reply by shmookles** [reply]
@mKeRix would be awesome if you could link up with the Home Assistant iOS companion App developer and see if they could merge your BLE companion capabilities into the main HA iOS App...

**Comment by gmalbert** [comment]
@mKeRix that's incredibly helpful, thank you. I see your concern about privacy circumvention. Could there also be a switch in the app to allow it to broadcast? I wonder if that might meet others' use cases.

**Reply by mKeRix** [reply]
It's certainly possible from a technical standpoint - feel free to open an issue for this at [mKeRix/room-assistant-companion-ios](https://github.com/mKeRix/room-assistant-companion-ios)!
